### PR TITLE
chore: make sure app is rehydrated before initializing unleash

### PR DESCRIPTION
**Pull request by MounirDhahri**
### Description

This PR resolves an issue where the user gets a different variant when the app state changes. 

As we were not testing if the app has finished rehydration, we were initializing unleash client without a userID. This led to sessions that can have a different variant from the variant specified to the user, which gets queried properly when the app is brought to the foreground.

**Before**

https://user-images.githubusercontent.com/11945712/215539486-02df67c4-908f-42c0-aa0e-a53d4b8df4d3.mov

**After**

https://user-images.githubusercontent.com/11945712/215539522-c9818ada-ce08-40cc-bcfe-66af90508992.mov

### PR Checklist

- [x] I tested my changes on **iOS** / **Android**.
- [x] I added screenshots or videos to illustrate my changes.
- [ ] I added Tests and Stories for my changes.
- [ ] I added an [app state migration].
- [ ] I hid my changes behind a [feature flag].
- [ ] I have prefixed changes that need to be tested during a release QA with **[NEEDS EXTERNAL QA]** on the changelog.

### To the reviewers 👀

- [ ] I would like **at least one** of the reviewers to run this PR on the simulator or device.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

-

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

- make sure app is rehydrated before initializing unleash - mounir

</details>

Need help with something? Have a look at our [docs], or get in touch with us.

[app state migration]: ../blob/main/docs/adding_state_migrations.md
[feature flag]: ../blob/main/docs/developing_a_feature.md
[docs]: ../blob/main/docs/README.md

### CODE REVIEW

**Review by lordkiz — APPROVED** [review]
LGTM 🎉

**Review by MrSltun — APPROVED** [review]
✨ 🛠️ 🌟

**Comment by MounirDhahri** [comment]
Merging this to get it tested and prepare it for release

File: src/app/utils/experiments/UnleashProvider.tsx [hunk 1]
```diff
@@ -12,29 +13,45 @@ export const UnleashContext = createContext<UnleashContext>({ lastUpdate: null }
...
-
-    client.on("error", () => {
-      console.error("Unleash error")
-    })
-
-    client.on("impression", () => {})
-
-    return () => {
-      client.stop()
+    if (isHydrated) {
```
Comment (pvinis) [inline src/app/utils/experiments/UnleashProvider.tsx:22]: oh i see! nice catch

**Pull request merged**